### DISCUSSION TITLE
💡 RFC: Official Nova.app extension

### DISCUSSION THREAD

**Discussion by dmythro** [withastro/language-tools#694] [Feature requests]
### Background & Motivation

[Nova.app](https://nova.app) by Panic is actually pretty good and I think I'll switch to it. It's fast, native, convenient and has almost everything I need for web dev.

And I see its community growing, so suggesting to have an official Astro extension. There's already [unofficial one](https://github.com/sciencefidelity/Nova-Astro) which helps, but it's outdated and as I can see not updated at all.

### Possible solutions

The existing extension could be a good start, I guess:
https://github.com/sciencefidelity/Nova-Astro

Author could be contacted by Astro team and ask what's the fastest way to get to speed.

### Help make it happen!

- [ ] I am willing to submit a PR to implement this change.
- [ ] I am willing to submit a PR to implement this change, but would need some guidance.
- [ ] I am not willing to submit a PR to implement this change.

**Comment by Princesseuh** [comment]
Last I checked (and it still seems to be the case from your link), the version of LSP they support is extremely outdated (as in, multiple years) in addition of having very very limited feature support, which made it hard to run the latest versions of our language server. 

The unofficial extension seems like it implement all the supported features, more or less. Was there any particular features you would like to see?

**Comment by dmythro** [comment]
Well, first of all, there are visual bugs like this:

<img width="663" alt="image" src="https://github.com/withastro/language-tools/assets/1391015/5fe66520-4bd7-40bd-b708-68f91fc04171">


Also, unpleasant things like no relevant typings etc:

<img width="260" alt="image" src="https://github.com/withastro/language-tools/assets/1391015/c9543845-fbfc-4ece-82db-1b2619ccd2cf">

Works better if specify `Astro.props as Props` manually, but I think it should be automatic, like [Astro documentation](https://docs.astro.build/en/guides/typescript/) states.

**Comment by dmythro** [comment]
I'm still a bit new to Astro, but I suppose the last commit to that extension in Oct 2022 says that some features aren't supported for sure taking into account that Astro had a major 2.0 release in 2023 :) 

Still, certainly better than nothing! And I use it now.

**Comment by dmythro** [comment]
VSCode extension handles this as expected and nicely formatted:

<img width="268" alt="image" src="https://github.com/withastro/language-tools/assets/1391015/aad5ed98-b124-42a0-8dec-4ed22bd9f2e4">

**Comment by Princesseuh** [comment]
> I suppose the last commit to that extension in Oct 2022 says that some features aren't supported for sure taking into account that Astro had a major 2.0 release in 2023 :)

Our language tooling is typically decoupled from Astro itself (It loads a fair amount of info from the project's Astro install), so there's usually no need for updating the language tooling. 

But yeah, it is running a fairly old version of the language-server at this point, running a newer version would improve some things. Though I'm not sure if it would fix the formatting issue in the hover, as far as I know we still send those the same way nowadays.

If Nova is able to run the latest version of our language-server, making an official extension for it is fairly trivial, if it cannot, it's unfortunately very unlikely we'll support it as it would require maintaining a language server just for them. This is ultimately low priority since the majority of our users is already covered (VS Code, Vim, JetBrains stuff), but I'll try to find time to check it out

**Comment by stonerl** [comment]
There is a new [Astro Extension for Nova](https://extensions.panic.com/extensions/com.johnlindop/com.johnlindop.nova-astro/). I am not sure how well it works, though.
